### ISSUE TITLE
be-memcached build problem

### ISSUE THREAD

**Issue by jcollie** [jpmens/mosquitto-auth-plug#336]
Trying to build latest git version:

```
CFLAGS:  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/src/ -I/lib/ -fPIC -Wall -Werror  -DBE_MEMCACHED -DBE_LDAP -DBE_FILES  -I/usr/local/include/libmemcached -I/src -DDEBUG=1 -I/usr/include
LDFLAGS: -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-hardened-ld  -L/usr/local/lib -L/lib/
LDADD:    -lmemcached -lldap -llber -L/usr/lib -lcrypto -lmosquitto

cc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/src/ -I/lib/ -fPIC -Wall -Werror  -DBE_MEMCACHED -DBE_LDAP -DBE_FILES  -I/usr/local/include/libmemcached -I/src -DDEBUG=1 -I/usr/include   -c -o auth-plug.o auth-plug.c
[ERROR   ] stderr: auth-plug.c: In function 'mosquitto_auth_plugin_init':
auth-plug.c:347:20: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
    (*bep)->getuser =  be_memcached_getuser;
                    ^
cc1: all warnings being treated as errors
make: *** [<builtin>: auth-plug.o] Error 1
```

**Comment by jpmens** [comment]
The backend now compiles; I cannot test, so I would very much appreciate if you could test this now.

**Comment by jcollie** [comment]
Works for me now, both with memcached enabled and disabled.

**Comment by jpmens** [comment]
With "works for me" do you mean "it compiles for me", or do you mean "auth plugin with memcached works and I've tested all the bells and whistles"? ;-)

**Comment by jcollie** [comment]
It compiles for me. I don't have a need for the memcached backend but was caught out by the fact that backends not listed in config.mk are enabled by the makefile.

**Comment by jpmens** [comment]
Addressed and this now fixes what @alteholz reported in #335
